shopfloor_base/models/shopfloor_app.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2021 Camptocamp SA (http://www.camptocamp.com)
+# @author Simone Orsi <simahawk@gmail.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+from functools import partial
+
+from odoo import api, fields, models
+from odoo.http import request
+
+from odoo.addons.base_rest.controllers.main import RestController
+from odoo.addons.base_rest.tools import _inspect_methods
+from odoo.addons.component.core import _component_databases
+
+
+def _process_endpoint(
+    self, app_id, service_name, service_method_name, *args, collection=None, **kwargs
+):
+    """Wrapper for  `_process_method` call.
+
+    Behavior is the same for the methods automatically generated by `rest.service.registration`.
+    """
+    collection = collection or request.env["shopfloor.app"].browse(app_id)
+    # TODO: in base_rest `*args` is passed based on on the type of route (eg: /<int:id>/update)
+    return self._process_method(
+        service_name, service_method_name, *args, collection=collection, params=kwargs
+    )
+
+
+RestController._process_endpoint = _process_endpoint
+
+
+class ShopfloorApp(models.Model):
+    """Backend for a Shopfloor app."""
+
+    _name = "shopfloor.app"
+    _inherit = "collection.base"
+    _description = "A Shopfloor application"
+
+    name = fields.Char(required=True, translate=True)
+    short_name = fields.Char(
+        required=True, translate=True, help="Needed for app manifest"
+    )
+    # Unique name
+    tech_name = fields.Char(required=True, index=True)
+    active = fields.Boolean(default=True)
+    category = fields.Selection(selection=[("", "None")])
+    api_route = fields.Char(
+        compute="_compute_api_route",
+        compute_sudo=True,
+        help="Base route for endpoints attached to this app, public version.",
+    )
+    api_route = fields.Char(
+        compute="_compute_api_route",
+        compute_sudo=True,
+        help="""
+        Base route for endpoints attached to this app,
+        internal controller-ready version.
+        """,
+    )
+    url = fields.Char(compute="_compute_url", help="Public URL to use the app.")
+    api_docs_url = fields.Char(compute="_compute_url", help="Public URL for api docs.")
+    auth_type = fields.Selection(
+        selection="_selection_auth_type", default="user_endpoint"
```

Comment:
The default value needs the module shopfloor_mobile_base_auth_user that is not a dependency

Reply:
no, this comes from `endpoint_route_handler` the way you implement it in the front is up to you.